test/test_unary_ufuncs.py

```diff
@@ -1,11 +1,226 @@
+import math
+from functools import partial
+from itertools import product, chain
+from numbers import Number
+import operator
+
+import unittest
+
 import torch
+import numpy as np
 
 from torch.testing._internal.common_utils import \
-    (TestCase, run_tests)
+    (TestCase, run_tests, torch_to_numpy_dtype_dict, suppress_warnings,
+     IS_WINDOWS)
 from torch.testing._internal.common_methods_invocations import \
     (unary_ufuncs)
 from torch.testing._internal.common_device_type import \
-    (instantiate_device_type_tests, ops, onlyOnCPUAndCUDA, skipCUDAIfRocm)
+    (instantiate_device_type_tests, ops, onlyOnCPUAndCUDA, skipCUDAIfRocm,
+     dtypes)
+from torch.testing import \
+    (floating_types_and)
+
+# Tensor generators and helpers
+
+# Interesting values and extremal values for different dtypes
+_unsigned_int_vals = (0, 1, 55, 127)
+_int_vals = (0, -1, 1, -55, 55, -127, 127)
+_small_float_vals = (0.,
+                     -.001, .001,
+                     -.25, .25,
+                     -1., 1.,
+                     -math.pi / 2, math.pi / 2,
+                     -math.pi + .00001, math.pi - .00001,
+                     -math.pi, math.pi,
+                     -math.pi - .00001, math.pi + .00001)
+_large_float_vals = (-501, 501,
+                     -1001.2, 1001.2,
+                     -13437.7, 13437.7,
+                     -4988429.2, 4988429.2,
+                     -1e20, 1e20)
+_float_extremals = (float('inf'), float('-inf'), float('nan'))
+
+def _make_tensor(size, device, dtype, *, domain=None):
+    if dtype is torch.bool:
+        return torch.tensor((True, False), device=device)
+
+    # Windows doesn't support torch.rand(bfloat16) on CUDA
+    if IS_WINDOWS and torch.device(device).type == 'cuda' and dtype is torch.bfloat16:
+        t = torch.rand(size, device=device, dtype=torch.float32).to(dtype)
+    else:
+        t = torch.rand(size, device=device, dtype=dtype)
```

Comment:
Int tensor generation is not very interesting - get a range of values (-100, 100)

Reply:
Is this even supported on integers? Add a unary ufunc that can actually accept ints